### PR TITLE
Screen player #307

### PR DESCRIPTION
**Pull request by osamaasifoak**
Hi @michaelcspeed , these are the things to review in this PR

- Audio player migrated to Riverpod
- Player screen

### CODE REVIEW

File: lib/view_model/player/player_viewmodel.dart [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  Future<void> downloadSessionAudio(String url, {String? fileName}) async {
+    try {
+      final downloadAudio = ref.read(downloaderRepositoryProvider);
+      await downloadAudio.downloadFile(
+        url,
+        name: fileName,
+        onReceiveProgress: (received, total) {
+          if (total != -1) {
+            downloadingProgress = (received / total * 100);
+            // (received / total * 100).toStringAsFixed(0) + '%';
```
Comment (michaelcspeed) [inline lib/view_model/player/player_viewmodel.dart]: unneeded comment?

File: lib/views/player/components/bottom_actions/bottom_action_component.dart [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            label:  StringConstants.DOWNLOAD.toUpperCase(),
+            onTap: () => {},
+          ),
+          width8,
+          if (sessionModel.hasBackgroundSound)
+            LabelsComponent(
+              label: StringConstants.SOUND.toUpperCase(),
+              onTap: () {
+                var location = GoRouter.of(context).location;
+                print(location + backgroundSounds);
```
Comment (michaelcspeed) [inline lib/views/player/components/bottom_actions/bottom_action_component.dart]: can remove this if not needed

File: lib/views/player/components/player_buttons_component.dart [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        SizedBox(width: 35),
+        _forwardButton(ref)
+      ],
+    );
+  }
+
+  InkWell _rewindButton(WidgetRef ref) {
+    return InkWell(
+      onTap: () =>
+          ref.read(skipAudioProvider(skip: SKIP_AUDIO.SKIP_BACWARD_10)),
```
Comment (michaelcspeed) [inline lib/views/player/components/player_buttons_component.dart]: spelling mistake :)

File: lib/views/player/components/player_buttons_component.dart [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    return InkWell(
+      onTap: () {
+        var _state = ref.watch(audioPlayPauseStateProvider.notifier).state;
+
+        ref.read(audioPlayPauseStateProvider.notifier).state =
+            _state == PLAY_PAUSE_AUDIO.PAUSE
+                ? PLAY_PAUSE_AUDIO.PLAY
+                : PLAY_PAUSE_AUDIO.PAUSE;
+      },
+      child: AnimatedCrossFade(
```
Comment (michaelcspeed) [inline lib/views/player/components/player_buttons_component.dart:61]: nice

**Review by michaelcspeed — CHANGES_REQUESTED** [review]

File: lib/views/player/components/player_buttons_component.dart (hunk 3 above)
Comment (osamaasifoak) [reply lib/views/player/components/player_buttons_component.dart]: Oops, let me fix this.

File: lib/view_model/player/player_viewmodel.dart (hunk 1 above)
Comment (osamaasifoak) [reply lib/view_model/player/player_viewmodel.dart]: Yes, I commented this line for a purpose, but now we no longer need it. I will remove this

**Pull request merged**